### PR TITLE
[Query] Update query_serversrc @open sesame 07/16 13:40

### PR DESCRIPTION
**Pull request by kimjh12**
Update serversrc element to serve remote tensor query
Currently working almost same as tcpserversrc
Create server socket and receive data from client on TCP

can test
```gst-launch-1.0 tensor_query_serversrc ! filesink location=temp.out```  
```echo hi | nc localhost 4953```  

Signed-off-by: Junhwan Kim <jejudo.kim@samsung.com>

---
**Self evaluation:**
1. Build test: [*]Passed [ ]Failed [ ]Skipped
2. Run test: [*]Passed [ ]Failed [ ]Skipped

### CODE REVIEW

File: gst/nnstreamer/tensor_query/tensor_query_common.c [hunk 1]
```diff
@@ -131,11 +138,29 @@ gst_tensor_query_socket_receive (GSocket * socket, GCancellable * cancellable,
...
+    if (need_alloc) {
+      out_mem = gst_allocator_alloc (NULL, read, NULL);
+    } else {
+      if (gst_buffer_get_size (outbuf) < read) {
+        gst_buffer_set_size (outbuf, read);
+      }
+      out_mem = gst_buffer_get_all_memory (outbuf);
+    }
+
+    nns_logi ("allocated buffer size: %lu", gst_buffer_get_size (outbuf));
```
Comment (dongju-chae) [inline gst/nnstreamer/tensor_query/tensor_query_common.c:154]: I think this log may not provide useful info. when `need_alloc` is true.

File: gst/nnstreamer/tensor_query/tensor_query_common.c [hunk 2]
```diff
@@ -93,6 +93,13 @@ gst_tensor_query_socket_receive (GSocket * socket, GCancellable * cancellable,
   gsize read;
   GError *err = NULL;
   GstMapInfo map;
+  GstMemory *out_mem;
+  gboolean need_alloc;
+
+  if (!socket || !outbuf) {
```
Comment (wooksong) [inline gst/nnstreamer/tensor_query/tensor_query_common.c:99]: ```socket``` would never be NULL and

```c
*outbuf = gst_buffer_new_and_alloc (0UL);
```
returns NULL according to https://github.com/genesi/gstreamer/blob/9d3066642e54c032200b919281f05427b1014527/gst/gstbuffer.c#L341-L356. Is it right? 

If you want to null-checking for ```outbuf```, do after line 340.

**Comment by myungjoo** [comment]
Off-Topic: in the commit message, you are supposed to describe the reason why you have added another argument and why you have added new conditions. Simply stating "- tcp server src element" is inappropriate.

**Comment by again4you** [comment]
What if this PR is in-progress state and you just submit it to share your code, then please use `Draft submits` or add `[WIP]` tag.

File: gst/nnstreamer/tensor_query/tensor_query_common.c (hunk 1 above)
Comment (kimjh12) [reply gst/nnstreamer/tensor_query/tensor_query_common.c]: Right. Moved it to log new allocated

File: gst/nnstreamer/tensor_query/tensor_query_common.c (hunk 2 above)
Comment (kimjh12) [reply gst/nnstreamer/tensor_query/tensor_query_common.c]: It seems it returns buffer object without memory allocated but NULL. (https://developer.gnome.org/gstreamer/stable/gstreamer-GstBuffer.html#gst-buffer-new-allocate)
While `*outbuf = gst_buffer_new_and_alloc (0UL);` is valid, I removed unnecessary logging. Thanks

**Review by dongju-chae — APPROVED** [review]
LGTM

**Pull request closed without merging**